pkg/kv/cosmosdb/store.go

```diff
@@ -458,7 +458,10 @@ func (e *EntriesIterator) Next() bool {
 // The reason we switch the batch size is to avoid issues like https://github.com/treeverse/lakeFS/issues/7864
 // as opposed to the exponential backoff approach in dynamoDB here we use a dynamic page size and let Cosmos DB manage paging.
 func (e *EntriesIterator) handleBatchSizeChange() error {
-	e.startKey = e.entry.Key
+	e.startKey = nil
+	if e.entry != nil {
+		e.startKey = e.entry.Key
+	}
```

Comment:
just a thought - but maybe we would want to log it in case this happens - just to help understand the case and how to reproduce

Reply:
That's a great idea!